Issues with SymmetricDeleteModel

# 1. SymmetricDeleteModel divides by 0

<!--- Please complete this template with required information for us to be able to reproduce it -->
<!--- If you are reporting an issue, failing to complete this template will result in closing the issue -->

<!--- Provide a general summary of the issue in the Title above -->

## Description

<!--- Provide a more detailed introduction to the issue itself, and why you consider it to be a bug -->

While trying to train a `SymmetricDeleteModel` using `SymmetricDeleteApproach`, an error occurs if the dataset only includes unique words (no duplicates).

This is due to how [`normalizeFrequencyValue`](https://github.com/JohnSnowLabs/spark-nlp/blob/1e22fe4bee10ed633651bf918f7cd490c7902848/src/main/scala/com/johnsnowlabs/nlp/annotators/spell/symmetric/SymmetricDeleteModel.scala#L198) is calculated.

`minFrequency` and `maxFrequency` are set during [`SymmetricDeleteApproach.train`](https://github.com/JohnSnowLabs/spark-nlp/blob/1e22fe4bee10ed633651bf918f7cd490c7902848/src/main/scala/com/johnsnowlabs/nlp/annotators/spell/symmetric/SymmetricDeleteApproach.scala#L180). If there are only unique words, both of them are set to `1`.

If `SymmetricDeleteModel.transform` of the resulting model is called division by 0 will occur:

```scala
val normalizedValue = (value - $(maxFrequency)).toDouble / ($(maxFrequency) - $(minFrequency)).toDouble
```

as `$(maxFrequency) - $(minFrequency)` is 0.

## Expected Behavior

<!--- Tell us what should happen -->

There should be no errors.

## Current Behavior

<!--- Tell us what happens instead of the expected behavior -->

There are errors and transform does not finish.

## Possible Solution

<!--- Not obligatory, but suggest a fix/reason for the bug, -->

Set `normalizedValue` to a fixed value, if `minFrequency` and `maxFrequency` are identical

## Steps to Reproduce

<!--- Provide a link to a live example or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->

1. Define prerequisites and SymmetricDeleteApproach
2. Fit to a dataset with only unique words
3. call transform

### Example Pipeline

<details>
  <summary>Click to expand</summary>

```scala
import com.johnsnowlabs.nlp.util.io.ResourceHelper
import org.apache.spark.sql.SparkSession

val spark: SparkSession = ResourceHelper.getActiveSparkSession

import spark.implicits._
import com.johnsnowlabs.nlp.base.DocumentAssembler
import com.johnsnowlabs.nlp.annotators.Tokenizer
import com.johnsnowlabs.nlp.annotators.spell.symmetric.SymmetricDeleteApproach
import org.apache.spark.ml.Pipeline

val documentAssembler = new DocumentAssembler()
  .setInputCol("text")
  .setOutputCol("document")

val tokenizer = new Tokenizer()
  .setInputCols("document")
  .setOutputCol("token")

val spellChecker = new SymmetricDeleteApproach()
  .setInputCols("token")
  .setOutputCol("spell")
  .setDictionary("src/test/resources/spell/words.txt")

val pipeline = new Pipeline().setStages(Array(
  documentAssembler,
  tokenizer,
  spellChecker
))

val data = Seq("spmetimes i wrrite wordz erong").toDF("text")
val pipelineModel = pipeline.fit(data)
val result = pipelineModel.transform(data)
result.select("spell.result").show(false)
```

</details>

## Stack Trace

<details>
  <summary>Click to expand</summary>

```
Failed to execute user defined function(HasSimpleAnnotate$$Lambda$1571/1119149914: (array<array<struct<annotatorType:string,begin:int,end:int,result:string,metadata:map<string,string>,embeddings:array<float>>>>) => array<struct<annotatorType:string,begin:int,end:int,result:string,metadata:map<string,string>,embeddings:array<float>>>)
org.apache.spark.SparkException: Failed to execute user defined function(HasSimpleAnnotate$$Lambda$1571/1119149914: (array<array<struct<annotatorType:string,begin:int,end:int,result:string,metadata:map<string,string>,embeddings:array<float>>>>) => array<struct<annotatorType:string,begin:int,end:int,result:string,metadata:map<string,string>,embeddings:array<float>>>)
	at org.apache.spark.sql.catalyst.expressions.ScalaUDF.eval(ScalaUDF.scala:1130)
	at org.apache.spark.sql.catalyst.expressions.Alias.eval(namedExpressions.scala:156)
	at org.apache.spark.sql.catalyst.expressions.InterpretedMutableProjection.apply(InterpretedMutableProjection.scala:83)
	at org.apache.spark.sql.catalyst.optimizer.ConvertToLocalRelation$$anonfun$apply$16.$anonfun$applyOrElse$73(Optimizer.scala:1514)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:238)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at scala.collection.TraversableLike.map(TraversableLike.scala:238)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:231)
	at scala.collection.immutable.List.map(List.scala:298)
	at org.apache.spark.sql.catalyst.optimizer.ConvertToLocalRelation$$anonfun$apply$16.applyOrElse(Optimizer.scala:1514)
	at org.apache.spark.sql.catalyst.optimizer.ConvertToLocalRelation$$anonfun$apply$16.applyOrElse(Optimizer.scala:1509)
	at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$transformDown$1(TreeNode.scala:315)
	at org.apache.spark.sql.catalyst.trees.CurrentOrigin$.withOrigin(TreeNode.scala:73)
	at org.apache.spark.sql.catalyst.trees.TreeNode.transformDown(TreeNode.scala:315)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.org$apache$spark$sql$catalyst$plans$logical$AnalysisHelper$$super$transformDown(LogicalPlan.scala:29)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.transformDown(AnalysisHelper.scala:162)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.transformDown$(AnalysisHelper.scala:160)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.transformDown(LogicalPlan.scala:29)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.transformDown(LogicalPlan.scala:29)
	at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$transformDown$3(TreeNode.scala:320)
	at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$mapChildren$1(TreeNode.scala:405)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapProductIterator(TreeNode.scala:243)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapChildren(TreeNode.scala:403)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapChildren(TreeNode.scala:356)
	at org.apache.spark.sql.catalyst.trees.TreeNode.transformDown(TreeNode.scala:320)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.org$apache$spark$sql$catalyst$plans$logical$AnalysisHelper$$super$transformDown(LogicalPlan.scala:29)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.transformDown(AnalysisHelper.scala:162)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.transformDown$(AnalysisHelper.scala:160)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.transformDown(LogicalPlan.scala:29)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.transformDown(LogicalPlan.scala:29)
	at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$transformDown$3(TreeNode.scala:320)
	at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$mapChildren$1(TreeNode.scala:405)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapProductIterator(TreeNode.scala:243)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapChildren(TreeNode.scala:403)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapChildren(TreeNode.scala:356)
	at org.apache.spark.sql.catalyst.trees.TreeNode.transformDown(TreeNode.scala:320)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.org$apache$spark$sql$catalyst$plans$logical$AnalysisHelper$$super$transformDown(LogicalPlan.scala:29)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.transformDown(AnalysisHelper.scala:162)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.transformDown$(AnalysisHelper.scala:160)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.transformDown(LogicalPlan.scala:29)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.transformDown(LogicalPlan.scala:29)
	at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$transformDown$3(TreeNode.scala:320)
	at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$mapChildren$1(TreeNode.scala:405)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapProductIterator(TreeNode.scala:243)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapChildren(TreeNode.scala:403)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapChildren(TreeNode.scala:356)
	at org.apache.spark.sql.catalyst.trees.TreeNode.transformDown(TreeNode.scala:320)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.org$apache$spark$sql$catalyst$plans$logical$AnalysisHelper$$super$transformDown(LogicalPlan.scala:29)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.transformDown(AnalysisHelper.scala:162)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.transformDown$(AnalysisHelper.scala:160)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.transformDown(LogicalPlan.scala:29)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.transformDown(LogicalPlan.scala:29)
	at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$transformDown$3(TreeNode.scala:320)
	at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$mapChildren$1(TreeNode.scala:405)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapProductIterator(TreeNode.scala:243)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapChildren(TreeNode.scala:403)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapChildren(TreeNode.scala:356)
	at org.apache.spark.sql.catalyst.trees.TreeNode.transformDown(TreeNode.scala:320)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.org$apache$spark$sql$catalyst$plans$logical$AnalysisHelper$$super$transformDown(LogicalPlan.scala:29)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.transformDown(AnalysisHelper.scala:162)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.transformDown$(AnalysisHelper.scala:160)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.transformDown(LogicalPlan.scala:29)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.transformDown(LogicalPlan.scala:29)
	at org.apache.spark.sql.catalyst.trees.TreeNode.transform(TreeNode.scala:304)
	at org.apache.spark.sql.catalyst.optimizer.ConvertToLocalRelation$.apply(Optimizer.scala:1509)
	at org.apache.spark.sql.catalyst.optimizer.ConvertToLocalRelation$.apply(Optimizer.scala:1508)
	at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$execute$2(RuleExecutor.scala:149)
	at scala.collection.IndexedSeqOptimized.foldLeft(IndexedSeqOptimized.scala:60)
	at scala.collection.IndexedSeqOptimized.foldLeft$(IndexedSeqOptimized.scala:68)
	at scala.collection.mutable.WrappedArray.foldLeft(WrappedArray.scala:38)
	at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$execute$1(RuleExecutor.scala:146)
	at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$execute$1$adapted(RuleExecutor.scala:138)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at org.apache.spark.sql.catalyst.rules.RuleExecutor.execute(RuleExecutor.scala:138)
	at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$executeAndTrack$1(RuleExecutor.scala:116)
	at org.apache.spark.sql.catalyst.QueryPlanningTracker$.withTracker(QueryPlanningTracker.scala:88)
	at org.apache.spark.sql.catalyst.rules.RuleExecutor.executeAndTrack(RuleExecutor.scala:116)
	at org.apache.spark.sql.execution.QueryExecution.$anonfun$optimizedPlan$1(QueryExecution.scala:82)
	at org.apache.spark.sql.catalyst.QueryPlanningTracker.measurePhase(QueryPlanningTracker.scala:111)
	at org.apache.spark.sql.execution.QueryExecution.$anonfun$executePhase$1(QueryExecution.scala:138)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:764)
	at org.apache.spark.sql.execution.QueryExecution.executePhase(QueryExecution.scala:138)
	at org.apache.spark.sql.execution.QueryExecution.optimizedPlan$lzycompute(QueryExecution.scala:79)
	at org.apache.spark.sql.execution.QueryExecution.optimizedPlan(QueryExecution.scala:79)
	at org.apache.spark.sql.execution.QueryExecution.$anonfun$writePlans$4(QueryExecution.scala:202)
	at org.apache.spark.sql.catalyst.plans.QueryPlan$.append(QueryPlan.scala:478)
	at org.apache.spark.sql.execution.QueryExecution.org$apache$spark$sql$execution$QueryExecution$$writePlans(QueryExecution.scala:202)
	at org.apache.spark.sql.execution.QueryExecution.toString(QueryExecution.scala:212)
	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$5(SQLExecution.scala:95)
	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:160)
	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:87)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:764)
	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:64)
	at org.apache.spark.sql.Dataset.withAction(Dataset.scala:3616)
	at org.apache.spark.sql.Dataset.head(Dataset.scala:2697)
	at org.apache.spark.sql.Dataset.take(Dataset.scala:2904)
	at org.apache.spark.sql.Dataset.getRows(Dataset.scala:300)
	at org.apache.spark.sql.Dataset.showString(Dataset.scala:337)
	at org.apache.spark.sql.Dataset.show(Dataset.scala:826)
	at org.apache.spark.sql.Dataset.show(Dataset.scala:803)
	at DocExamplesTest.$anonfun$new$1(DocExamplesTest.scala:42)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
	at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
	at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
	at org.scalatest.Transformer.apply(Transformer.scala:22)
	at org.scalatest.Transformer.apply(Transformer.scala:20)
	at org.scalatest.FlatSpecLike$$anon$1.apply(FlatSpecLike.scala:1682)
	at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
	at org.scalatest.TestSuite.withFixture$(TestSuite.scala:195)
	at org.scalatest.FlatSpec.withFixture(FlatSpec.scala:1685)
	at org.scalatest.FlatSpecLike.invokeWithFixture$1(FlatSpecLike.scala:1680)
	at org.scalatest.FlatSpecLike.$anonfun$runTest$1(FlatSpecLike.scala:1692)
	at org.scalatest.SuperEngine.runTestImpl(Engine.scala:289)
	at org.scalatest.FlatSpecLike.runTest(FlatSpecLike.scala:1692)
	at org.scalatest.FlatSpecLike.runTest$(FlatSpecLike.scala:1674)
	at org.scalatest.FlatSpec.runTest(FlatSpec.scala:1685)
	at org.scalatest.FlatSpecLike.$anonfun$runTests$1(FlatSpecLike.scala:1750)
	at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:396)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:384)
	at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:379)
	at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:461)
	at org.scalatest.FlatSpecLike.runTests(FlatSpecLike.scala:1750)
	at org.scalatest.FlatSpecLike.runTests$(FlatSpecLike.scala:1749)
	at org.scalatest.FlatSpec.runTests(FlatSpec.scala:1685)
	at org.scalatest.Suite.run(Suite.scala:1147)
	at org.scalatest.Suite.run$(Suite.scala:1129)
	at org.scalatest.FlatSpec.org$scalatest$FlatSpecLike$$super$run(FlatSpec.scala:1685)
	at org.scalatest.FlatSpecLike.$anonfun$run$1(FlatSpecLike.scala:1795)
	at org.scalatest.SuperEngine.runImpl(Engine.scala:521)
	at org.scalatest.FlatSpecLike.run(FlatSpecLike.scala:1795)
	at org.scalatest.FlatSpecLike.run$(FlatSpecLike.scala:1793)
	at org.scalatest.FlatSpec.run(FlatSpec.scala:1685)
	at org.scalatest.tools.SuiteRunner.run(SuiteRunner.scala:45)
	at org.scalatest.tools.Runner$.$anonfun$doRunRunRunDaDoRunRun$13(Runner.scala:1340)
	at org.scalatest.tools.Runner$.$anonfun$doRunRunRunDaDoRunRun$13$adapted(Runner.scala:1334)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at org.scalatest.tools.Runner$.doRunRunRunDaDoRunRun(Runner.scala:1334)
	at org.scalatest.tools.Runner$.$anonfun$runOptionallyWithPassFailReporter$24(Runner.scala:1031)
	at org.scalatest.tools.Runner$.$anonfun$runOptionallyWithPassFailReporter$24$adapted(Runner.scala:1010)
	at org.scalatest.tools.Runner$.withClassLoaderAndDispatchReporter(Runner.scala:1500)
	at org.scalatest.tools.Runner$.runOptionallyWithPassFailReporter(Runner.scala:1010)
	at org.scalatest.tools.Runner$.run(Runner.scala:850)
	at org.scalatest.tools.Runner.run(Runner.scala)
	at org.jetbrains.plugins.scala.testingSupport.scalaTest.ScalaTestRunner.runScalaTest2or3(ScalaTestRunner.java:38)
	at org.jetbrains.plugins.scala.testingSupport.scalaTest.ScalaTestRunner.main(ScalaTestRunner.java:25)
Caused by: java.lang.NumberFormatException
	at java.math.BigDecimal.<init>(BigDecimal.java:497)
	at java.math.BigDecimal.<init>(BigDecimal.java:827)
	at scala.math.BigDecimal$.decimal(BigDecimal.scala:53)
	at scala.math.BigDecimal$.apply(BigDecimal.scala:250)
	at com.johnsnowlabs.nlp.annotators.spell.symmetric.SymmetricDeleteModel.normalizeFrequencyValue(SymmetricDeleteModel.scala:199)
	at com.johnsnowlabs.nlp.annotators.spell.symmetric.SymmetricDeleteModel.getScoreFrequency(SymmetricDeleteModel.scala:188)
	at com.johnsnowlabs.nlp.annotators.spell.symmetric.SymmetricDeleteModel.getDictionarySuggestions(SymmetricDeleteModel.scala:176)
	at com.johnsnowlabs.nlp.annotators.spell.symmetric.SymmetricDeleteModel.getSuggestedCorrections(SymmetricDeleteModel.scala:166)
	at com.johnsnowlabs.nlp.annotators.spell.symmetric.SymmetricDeleteModel.checkSpellWord(SymmetricDeleteModel.scala:103)
	at com.johnsnowlabs.nlp.annotators.spell.symmetric.SymmetricDeleteModel.$anonfun$annotate$1(SymmetricDeleteModel.scala:84)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:238)
	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
	at scala.collection.TraversableLike.map(TraversableLike.scala:238)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:231)
	at scala.collection.AbstractTraversable.map(Traversable.scala:108)
	at com.johnsnowlabs.nlp.annotators.spell.symmetric.SymmetricDeleteModel.annotate(SymmetricDeleteModel.scala:83)
	at com.johnsnowlabs.nlp.HasSimpleAnnotate.$anonfun$dfAnnotate$1(HasSimpleAnnotate.scala:24)
	at org.apache.spark.sql.catalyst.expressions.ScalaUDF.$anonfun$f$2(ScalaUDF.scala:157)
	at org.apache.spark.sql.catalyst.expressions.ScalaUDF.eval(ScalaUDF.scala:1127)
	... 146 more

```

</details>

## Your Environment

<!--- Include as many relevant details about the environment you experienced the bug in -->

- Spark NLP version `sparknlp.version()`: 3.1.1
- Apache NLP version `spark.version`: 3.0.2
- Java version `java -version`: 1.8
- Setup and installation (Pypi, Conda, Maven, etc.): Ran test from 311-release-candidate branch
- Operating System and version: Manjaro Linux x86_64

<!--- Please complete this template with required information for us to be able to reproduce it -->
<!--- If you are reporting an issue, failing to complete this template will result in closing the issue -->

# 2. Default SymmetricDeleteModel

## Description

<!--- Provide a more detailed introduction to the issue itself, and why you consider it to be a bug -->

Running a pipeline with the default pretrained model results in `OutOfMemoryError: Java heap space` locally, as well as on Databricks

## Stack Trace

<details>
  <summary>Click to expand</summary>

```
An exception or error caused a run to abort: Java heap space
java.lang.OutOfMemoryError: Java heap space
	at java.util.IdentityHashMap.resize(IdentityHashMap.java:472)
	at java.util.IdentityHashMap.put(IdentityHashMap.java:441)
	at org.apache.spark.util.SizeEstimator$SearchState.enqueue(SizeEstimator.scala:176)
	at org.apache.spark.util.SizeEstimator$.$anonfun$visitSingleObject$1(SizeEstimator.scala:227)
	at org.apache.spark.util.SizeEstimator$.$anonfun$visitSingleObject$1$adapted(SizeEstimator.scala:226)
	at org.apache.spark.util.SizeEstimator$$$Lambda$1478/1798667958.apply(Unknown Source)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at org.apache.spark.util.SizeEstimator$.visitSingleObject(SizeEstimator.scala:226)
	at org.apache.spark.util.SizeEstimator$.estimate(SizeEstimator.scala:203)
	at org.apache.spark.util.SizeEstimator$.estimate(SizeEstimator.scala:70)
	at org.apache.spark.util.collection.SizeTracker.takeSample(SizeTracker.scala:78)
	at org.apache.spark.util.collection.SizeTracker.afterUpdate(SizeTracker.scala:70)
	at org.apache.spark.util.collection.SizeTracker.afterUpdate$(SizeTracker.scala:67)
	at org.apache.spark.util.collection.SizeTrackingVector.$plus$eq(SizeTrackingVector.scala:31)
	at org.apache.spark.storage.memory.DeserializedValuesHolder.storeValue(MemoryStore.scala:665)
	at org.apache.spark.storage.memory.MemoryStore.putIterator(MemoryStore.scala:222)
	at org.apache.spark.storage.memory.MemoryStore.putIteratorAsValues(MemoryStore.scala:299)
	at org.apache.spark.storage.BlockManager.$anonfun$doPutIterator$1(BlockManager.scala:1371)
	at org.apache.spark.storage.BlockManager$$Lambda$1463/1282035709.apply(Unknown Source)
	at org.apache.spark.storage.BlockManager.org$apache$spark$storage$BlockManager$$doPut(BlockManager.scala:1298)
	at org.apache.spark.storage.BlockManager.doPutIterator(BlockManager.scala:1362)
	at org.apache.spark.storage.BlockManager.putIterator(BlockManager.scala:1218)
	at org.apache.spark.storage.BlockManager.putSingle(BlockManager.scala:1687)
	at org.apache.spark.broadcast.TorrentBroadcast.writeBlocks(TorrentBroadcast.scala:133)
	at org.apache.spark.broadcast.TorrentBroadcast.<init>(TorrentBroadcast.scala:91)
	at org.apache.spark.broadcast.TorrentBroadcastFactory.newBroadcast(TorrentBroadcastFactory.scala:35)
	at org.apache.spark.broadcast.BroadcastManager.newBroadcast(BroadcastManager.scala:77)
	at org.apache.spark.SparkContext.broadcast(SparkContext.scala:1486)
	at com.johnsnowlabs.nlp.serialization.Feature.$anonfun$setValue$1(Feature.scala:87)
	at com.johnsnowlabs.nlp.serialization.Feature$$Lambda$2086/1648992397.apply(Unknown Source)
	at scala.Option.map(Option.scala:230)
	at com.johnsnowlabs.nlp.serialization.Feature.setValue(Feature.scala:87)
```

</details>

## Expected Behavior

<!--- Tell us what should happen -->

There should be no errors.

## Current Behavior

<!--- Tell us what happens instead of the expected behavior -->

There are errors and transform does not finish.

## Steps to Reproduce

<!--- Provide a link to a live example or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->

1. Define prerequisites and `SymmetricDeleteModel.pretrained()`
2. call transform

### Example Pipeline

<details>
  <summary>Click to expand</summary>

```scala
import spark.implicits._
import com.johnsnowlabs.nlp.base.DocumentAssembler
import com.johnsnowlabs.nlp.annotators.Tokenizer
import com.johnsnowlabs.nlp.annotators.spell.symmetric.SymmetricDeleteModel
import org.apache.spark.ml.Pipeline

val documentAssembler = new DocumentAssembler()
  .setInputCol("text")
  .setOutputCol("document")

val tokenizer = new Tokenizer()
  .setInputCols("document")
  .setOutputCol("token")

val spellChecker = SymmetricDeleteModel.pretrained()
  .setInputCols("token")
  .setOutputCol("spell")

val pipeline = new Pipeline().setStages(Array(
  documentAssembler,
  tokenizer,
  spellChecker
))

val data = Seq("spmetimes i wrrite wordz erong.").toDF("text")
val result = pipeline.fit(data).transform(data)
result.select("spell.result").show(false)
```

</details>

## Your Environment

<!--- Include as many relevant details about the environment you experienced the bug in -->

- Locally same as above
- Databricks:
  - 8.1 (includes Apache Spark 3.1.1, Scala 2.12)
  - ```
    spark.kryoserializer.buffer.max 2000M
    spark.serializer org.apache.spark.serializer.KryoSerializer
    spark.databricks.delta.preview.enabled true
    ```
  - com.johnsnowlabs.nlp:spark-nlp_2.12:3.0.2

--- a/src/main/scala/com/johnsnowlabs/nlp/annotators/spell/symmetric/SymmetricDeleteModel.scala
+++ b/src/main/scala/com/johnsnowlabs/nlp/annotators/spell/symmetric/SymmetricDeleteModel.scala
@@ -1,3 +1,20 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package com.johnsnowlabs.nlp.annotators.spell.symmetric
 
 import com.johnsnowlabs.nlp.annotators.spell.util.Utilities
@@ -243,7 +260,7 @@ class SymmetricDeleteModel(override val uid: String) extends AnnotatorModel[Symm
     if (value < $(minFrequency)) {
       return 0
     }
-    if ($(maxFrequency) == $(minFrequency)){
+    if ($(maxFrequency) == $(minFrequency)) {
       return 1
     }
     val normalizedValue = (value - $(maxFrequency)).toDouble / ($(maxFrequency) - $(minFrequency)).toDouble

--- a/src/main/scala/com/johnsnowlabs/nlp/annotators/spell/symmetric/SymmetricDeleteModel.scala
+++ b/src/main/scala/com/johnsnowlabs/nlp/annotators/spell/symmetric/SymmetricDeleteModel.scala
@@ -243,6 +243,9 @@ class SymmetricDeleteModel(override val uid: String) extends AnnotatorModel[Symm
     if (value < $(minFrequency)) {
       return 0
     }
+    if ($(maxFrequency) == $(minFrequency)){
+      return 1
+    }
     val normalizedValue = (value - $(maxFrequency)).toDouble / ($(maxFrequency) - $(minFrequency)).toDouble
     BigDecimal(normalizedValue).setScale(4, BigDecimal.RoundingMode.HALF_UP).toDouble
   }

--- a/src/test/scala/com/johnsnowlabs/nlp/annotators/spell/symmetric/SymmetricDeleteBehaviors.scala
+++ b/src/test/scala/com/johnsnowlabs/nlp/annotators/spell/symmetric/SymmetricDeleteBehaviors.scala
@@ -314,4 +314,31 @@ trait SymmetricDeleteBehaviors { this: FlatSpec =>
     }
   }
 
+  def testTrainUniqueWordsOnly(): Unit = {
+    "Using a corpus with only unique words" should "successfully correct words" taggedAs FastTest in {
+
+      val spell = new SymmetricDeleteApproach()
+        .setInputCols("token")
+        .setOutputCol("spell")
+        .setDictionary("src/test/resources/spell/words.txt")
+
+      val finisher = new Finisher()
+        .setInputCols("spell")
+
+      val pipeline = new Pipeline()
+        .setStages(Array(
+          documentAssembler,
+          tokenizer,
+          spell,
+          finisher
+        ))
+
+      val uniqueWordsTrain = Seq("Only unique words.").toDF("text")
+      val model = pipeline.fit(uniqueWordsTrain)
+
+      val result = model.transform(predictionDataSet)
+      assert(result.isInstanceOf[DataFrame])
+      result.select("finished_spell").show(false)
+    }
+  }
 }

--- a/src/test/scala/com/johnsnowlabs/nlp/annotators/spell/symmetric/SymmetricDeleteBehaviors.scala
+++ b/src/test/scala/com/johnsnowlabs/nlp/annotators/spell/symmetric/SymmetricDeleteBehaviors.scala
@@ -1,3 +1,20 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package com.johnsnowlabs.nlp.annotators.spell.symmetric
 
 import com.johnsnowlabs.nlp.SparkAccessor.spark.implicits._
@@ -11,7 +28,8 @@ import org.apache.spark.sql.{DataFrame, Dataset}
 import org.scalatest._
 
 
-trait SymmetricDeleteBehaviors { this: FlatSpec =>
+trait SymmetricDeleteBehaviors {
+  this: FlatSpec =>
 
   private val trainDataSet = AnnotatorBuilder.getTrainingDataSet("src/test/resources/spell/sherlockholmes.txt")
   private val predictionDataSet = ContentProvider.parquetData.limit(500)
@@ -63,21 +81,21 @@ trait SymmetricDeleteBehaviors { this: FlatSpec =>
     pipeline.fit(emptyDataSet).transform(corpusDataSet)
   }
 
-  def testSimpleCheck(wordAnswer: Seq[(String, String)]): Unit ={
-     it should "successfully correct a misspell" taggedAs FastTest in {
-       val trainDataSet = getTrainDataSet("src/test/resources/spell/sherlockholmes.txt")
-       val spellChecker = new SymmetricDeleteApproach()
-         .setInputCols("token")
-         .setOutputCol("spell")
-         .fit(trainDataSet)
-       val misspell = wordAnswer.head._1
+  def testSimpleCheck(wordAnswer: Seq[(String, String)]): Unit = {
+    it should "successfully correct a misspell" taggedAs FastTest in {
+      val trainDataSet = getTrainDataSet("src/test/resources/spell/sherlockholmes.txt")
+      val spellChecker = new SymmetricDeleteApproach()
+        .setInputCols("token")
+        .setOutputCol("spell")
+        .fit(trainDataSet)
+      val misspell = wordAnswer.head._1
 
       val correction = spellChecker.checkSpellWord(misspell)
       assert(correction._1 == wordAnswer.head._2)
     }
   }
 
-  def testSeveralChecks(wordAnswer:  Seq[(String, String)]): Unit = {
+  def testSeveralChecks(wordAnswer: Seq[(String, String)]): Unit = {
     s"symspell checker " should s"successfully correct several misspells" taggedAs FastTest in {
 
       val trainDataSet = getTrainDataSet("src/test/resources/spell/sherlockholmes.txt")
@@ -86,7 +104,7 @@ trait SymmetricDeleteBehaviors { this: FlatSpec =>
         .setOutputCol("spell")
         .fit(trainDataSet)
 
-      wordAnswer.foreach( wa => {
+      wordAnswer.foreach(wa => {
         val misspell = wa._1
         val correction = spellChecker.checkSpellWord(misspell)
         assert(correction._1 == wa._2)
@@ -127,7 +145,7 @@ trait SymmetricDeleteBehaviors { this: FlatSpec =>
         ))
 
       val model = pipeline.fit(trainDataSet)
-      
+
       assert(model.transform(predictionDataSet).isInstanceOf[DataFrame])
     }
   }
@@ -163,11 +181,11 @@ trait SymmetricDeleteBehaviors { this: FlatSpec =>
         var correctedData = model.transform(predictionDataSet)
         correctedData = correctedData.withColumn("prediction",
           when(col("word") === col("finished_spell"), 1).otherwise(0))
-        val rightCorrections = correctedData.filter(col("prediction")===1).count()
-        val wrongCorrections = correctedData.filter(col("prediction")===0).count()
+        val rightCorrections = correctedData.filter(col("prediction") === 1).count()
+        val wrongCorrections = correctedData.filter(col("prediction") === 0).count()
         printf("Right Corrections: %d \n", rightCorrections)
         printf("Wrong Corrections: %d \n", wrongCorrections)
-        val accuracy = rightCorrections.toFloat/(rightCorrections+wrongCorrections).toFloat
+        val accuracy = rightCorrections.toFloat / (rightCorrections + wrongCorrections).toFloat
         printf("Accuracy: %f\n", accuracy)
       }
     }
@@ -200,23 +218,23 @@ trait SymmetricDeleteBehaviors { this: FlatSpec =>
   def testEmptyDataset(): Unit = {
     it should "rise an error message" taggedAs FastTest in {
 
-    val spell = new SymmetricDeleteApproach()
-      .setInputCols(Array("token"))
-      .setDictionary("src/test/resources/spell/words.txt")
-      .setOutputCol("spell")
+      val spell = new SymmetricDeleteApproach()
+        .setInputCols(Array("token"))
+        .setDictionary("src/test/resources/spell/words.txt")
+        .setOutputCol("spell")
 
-    val finisher = new Finisher()
-      .setInputCols("spell")
-      .setOutputAsArray(false)
-      .setIncludeMetadata(false)
+      val finisher = new Finisher()
+        .setInputCols("spell")
+        .setOutputAsArray(false)
+        .setIncludeMetadata(false)
 
-    val pipeline = new Pipeline()
-      .setStages(Array(
-        documentAssembler,
-        tokenizer,
-        spell,
-        finisher
-      ))
+      val pipeline = new Pipeline()
+        .setStages(Array(
+          documentAssembler,
+          tokenizer,
+          spell,
+          finisher
+        ))
 
       import SparkAccessor.spark.implicits._
 
@@ -278,11 +296,11 @@ trait SymmetricDeleteBehaviors { this: FlatSpec =>
 
     val predictionDataSet = words.toDF("text")
     val model = pipeline.fit(trainDataSet)
-    model.transform(predictionDataSet).select("spell").show(1,false)
+    model.transform(predictionDataSet).select("spell").show(1, false)
   }
 
   def trainSpellCheckerModelFromFit(): Unit = {
-    "" should "trained a model from fit when dataset with array annotation is used"  in {
+    "" should "trained a model from fit when dataset with array annotation is used" in {
       val trainDataSet = getTrainDataSet("src/test/resources/spell/sherlockholmes.txt")
 
       val spell = new SymmetricDeleteApproach()
@@ -298,7 +316,7 @@ trait SymmetricDeleteBehaviors { this: FlatSpec =>
   }
 
   def raiseErrorWhenWrongColumnIsSent(): Unit = {
-    "" should "raise an error when dataset without array annotation is used"  in {
+    "" should "raise an error when dataset without array annotation is used" in {
       val trainDataSet = AnnotatorBuilder.getTrainingDataSet("src/test/resources/spell/sherlockholmes.txt")
       val expectedErrorMessage = "Train dataset must have an array annotation type column"
       val spell = new SymmetricDeleteApproach()

--- a/src/test/scala/com/johnsnowlabs/nlp/annotators/spell/symmetric/SymmetricDeleteModelTestSpec.scala
+++ b/src/test/scala/com/johnsnowlabs/nlp/annotators/spell/symmetric/SymmetricDeleteModelTestSpec.scala
@@ -18,45 +18,47 @@ class SymmetricDeleteModelTestSpec extends FlatSpec with SymmetricDeleteBehavior
 
   "a simple isolated symmetric spell checker with noisy word" should behave like testSimpleCheck(
     Seq(("CARDIOVASCULAR:", "CARDIOVASCULAR:")))
-/*
-  "an isolated symmetric spell checker " should behave like testSeveralChecks(Seq(
-    ("problex", "problem"),
-    ("contende", "continue"),
-    ("pronounciation", "pronounciation"),
-    ("localy", "local"),
-    ("compair", "company")
-  ))
 
-  "a symmetric spell checker " should behave like testAccuracyChecks(Seq(
-    ("mral", "meal"),
-    ("delicatly", "delicately"),
-    ("efusive", "effusive"),
-    ("lauging", "laughing"),
-    ("juuuuuuuuuuuuuuuussssssssssttttttttttt", "just"),
-    ("screeeeeeeewed", "screwed"),
-    ("readampwritepeaceee", "readampwritepeaceee")
-  ))
+  it should behave like testTrainUniqueWordsOnly()
+  /*
+    "an isolated symmetric spell checker " should behave like testSeveralChecks(Seq(
+      ("problex", "problem"),
+      ("contende", "continue"),
+      ("pronounciation", "pronounciation"),
+      ("localy", "local"),
+      ("compair", "company")
+    ))
 
-  "a good sized dataframe with Spark pipeline" should behave like testBigPipeline
+    "a symmetric spell checker " should behave like testAccuracyChecks(Seq(
+      ("mral", "meal"),
+      ("delicatly", "delicately"),
+      ("efusive", "effusive"),
+      ("lauging", "laughing"),
+      ("juuuuuuuuuuuuuuuussssssssssttttttttttt", "just"),
+      ("screeeeeeeewed", "screwed"),
+      ("readampwritepeaceee", "readampwritepeaceee")
+    ))
 
-  "a good sized dataframe with Spark pipeline and spell checker dictionary" should behave like testBigPipelineDict
+    "a good sized dataframe with Spark pipeline" should behave like testBigPipeline
 
-  "a loaded model " should behave like testLoadModel()
+    "a good sized dataframe with Spark pipeline and spell checker dictionary" should behave like testBigPipelineDict
 
-  "a symmetric spell checker with empty dataset" should behave like testEmptyDataset
+    "a loaded model " should behave like testLoadModel()
 
-  "A symmetric spell checker trained from fit" should behave like trainFromFitSpellChecker(Seq(
-    "mral",
-    "delicatly",
-    "efusive",
-    "lauging",
-    "juuuuuuuuuuuuuuuussssssssssttttttttttt",
-    "screeeeeeeewed",
-    "readampwritepeaceee"
-  ))
+    "a symmetric spell checker with empty dataset" should behave like testEmptyDataset
 
-  it should behave like trainSpellCheckerModelFromFit
+    "A symmetric spell checker trained from fit" should behave like trainFromFitSpellChecker(Seq(
+      "mral",
+      "delicatly",
+      "efusive",
+      "lauging",
+      "juuuuuuuuuuuuuuuussssssssssttttttttttt",
+      "screeeeeeeewed",
+      "readampwritepeaceee"
+    ))
 
-  it should behave like raiseErrorWhenWrongColumnIsSent
-*/
+    it should behave like trainSpellCheckerModelFromFit
+
+    it should behave like raiseErrorWhenWrongColumnIsSent
+  */
 }

--- a/src/test/scala/com/johnsnowlabs/nlp/annotators/spell/symmetric/SymmetricDeleteModelTestSpec.scala
+++ b/src/test/scala/com/johnsnowlabs/nlp/annotators/spell/symmetric/SymmetricDeleteModelTestSpec.scala
@@ -1,3 +1,20 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package com.johnsnowlabs.nlp.annotators.spell.symmetric
 
 

